src/buckets/components/createBucketForm/MeasurementSchemaSection.tsx

```diff
@@ -188,10 +176,14 @@ const AddingPanel: FC<AddingProps> = ({
   // since the mini file dnd component calls this method,
   // then calls setError(false).  so; if we call setError(true) then it gets immediately
   // overridden.  with the propagation, the error state only gets called once properly.
-  const handleUploadFile = (contents: string, fileName: string) => {
+  const handleFileUpload = (
+    contents: string,
+    isCsv: boolean,
```

Comment:
please change this to an enum or a union type. Boolean arguments don't communicate the method's intent nearly as well as a literal value.

```ts
const handleFileUpload = (
  contents: string,
  fileType: DownloadTypes
  fileName: string
)
```

this enumerates the possible values in the method signature. It also describes what is happening in the method call and makes it more self-documenting.

```ts
handleFileUpload(contents, true, 'mydata')
handleFileUpload(contents, false, 'mydata2')
```
vs
```ts
handleFileUpload(contents, 'csv', 'mydata')
handleFileUpload(contents, 'json', 'mydata2')
```